src/lightning/pytorch/trainer/trainer.py

```diff
@@ -1532,18 +1532,15 @@ def num_training_batches(self) -> Union[int, float]:
         return self.fit_loop.max_batches
```

Comment:
Shouldn't num_training_batches also return a list? Or not because it only supports `max_size_cycle`? If possible, it would be useful to make the change now together with the other properties or else we will need to change it again in the future

Reply:
IMO no, this would be too big of a breaking change. People call this to configure their learning rate schedulers and so on. It not returning an int is confusing.

Reply:
Wouldn't that logic then also motivate not returning lists for the evaluation and prediction properties?

Reply:
The properties returned lists before this PR as well. Regardless of the decision I don't see why this should be a responsibility of this PR.